direct/src/showbase/CMakeLists.txt

```diff
@@ -1,5 +1,15 @@
-add_component_library(p3showbase SYMBOL BUILDING_DIRECT_SHOWBASE
-  showBase.cxx showBase.h)
+set(P3SHOWBASE_FILES
+    showBase.cxx
+    showBase.h)
+set(P3SHOWBASE_EXTRA_ARGS "")
+if(IS_OSX) 
+  list(APPEND P3SHOWBASE_FILES showBase_assist.mm)
+endif()
+
+add_component_library(p3showbase SYMBOL BUILDING_DIRECT_SHOWBASE ${P3SHOWBASE_FILES})
+if(APPLE)
```

Comment:
The first conditional is for something that's really only applicable for macOS but not any other Apple platform, while the second would be potentially useful for both macOS and iOS. I realize macOS is the only Apple platform Panda supports at the moment, but I figured the distinction would be useful down the road when/if iOS starts being supported.

Reply:
[CMake uses APPLE to indicate macOS](https://cmake.org/cmake/help/v3.0/variable/APPLE.html) and the `IS_OSX` is the preprocessor macro indicating the same. They're largely interchangeable.

I'm wondering if we should remove `IS_OSX` from CMake (detect it the same way WIN32_VC is detected) and use `APPLE` everywhere in CMake.

Reply:
@treamology but it's silly to use different variables here, because the only reason the `-ObjC++` flag is passed is because an `.mm` (Objective-C++) file is included.

IIRC `IS_OSX` is a remnant from the ppremake days, and I think that ppremake used to also define `IS_OSX` for the iOS build of Panda.  So I suggest that we just use `APPLE` for both here.